File: lscene.js
//JS code by Luke Lincoln

function LScene()
{
	this.objects = [];
	this.num_integrates = 10;
}

LScene.prototype.add_object = function(pnts, is_fixed, props)
{
	this.objects.push(new LObj(pnts, is_fixed, props));
};

LScene.prototype.step = function(t)
{
	var i = 0, j;
	var td = t / this.num_integrates;
	for(; i < this.objects.length; i++)
		for(j = 0; j < this.num_integrates; j++)
			this.objects[i].step(td);
};

LScene.prototype.apply_physics = function()
{
	 var i = 0, j;
	 for(; i < this.objects.length-1; i++)
	 {
		 for(j = i+1; j < this.objects.length; j++)
		 {
			var ob1 = this.objects[i];
			var ob2 = this.objects[j];
			if(ob1.fixed && ob2.fixed) continue;
			var s1 = ob1.get_shape();
			var s2 = ob2.get_shape();
			var col = s1.collision_detection(s2);
			//for collision i need: 
			/*
				whether there was a collision:
				a list of collisions:
					the point of collision
					the penetration or distance
					the normal of the surface involved
			*/
			if(col.collision)
			{
				var k = 0;
				var n = col.list.length;
				for(; k < n; k++)
				{
					ll_pos_correct(ob1, ob2, col.list[k].dist, col.list[k].norm.clone().inv());
					ll_resolve_collision(ob1, ob2, col.list[k].norm.clone().inv(), col.list[k].point.clone(), n);
				}
				
			}
		 }
	 }
};

File: java implementation/seperated files/LScene.java
import java.util.ArrayList;


public class LScene {


	ArrayList<LObj> objects;
	int num_integrates;
	
	public LScene()
	{
		this.objects = new ArrayList<LObj>();
		this.num_integrates = 100;
	}
	
	public void add_object(P2i[] pnts, boolean is_fixed)
	{
		this.objects.add(new LObj(pnts, is_fixed));
	}
	
	public void add_object(P2i[] pnts, boolean is_fixed, P2i pos)
	{
		
		LObj ob = new LObj(pnts, is_fixed);
		ob.pos = pos.cpy();
		this.objects.add(ob);
	}
	
	public void add_object(P2i[] pnts, boolean is_fixed, P2i pos, P2i vel, double ang)
	{
		LObj ob = new LObj(pnts, is_fixed);
		ob.pos = pos.cpy();
		ob.vel = vel.cpy();
		ob.ang_vel = ang;
		this.objects.add(ob);
	}
	
	public void add_object(P2i[] pnts, boolean is_fixed, P2i pos, double orient)
	{
		LObj ob = new LObj(pnts, is_fixed);
		ob.pos = pos.cpy();
		ob.orient = orient;
		
		this.objects.add(ob);
	}
	
	public void step(double t)
	{
		double td = t / (double)this.num_integrates;
		for(int i = 0; i < this.objects.size(); i++)
			for(int j = 0; j < this.num_integrates; j++)
				this.objects.get(i).step(td);
	}
	
	public void apply_physics()
	{
		 for(int i = 0; i < this.objects.size()-1; i++)
		 {
			 for(int j = i+1; j < this.objects.size(); j++)
			 {
				LObj ob1 = this.objects.get(i);
				LObj ob2 = this.objects.get(j);
				if(ob1.fixed && ob2.fixed) continue;
				LShape s1 = ob1.get_shape();
				LShape s2 = ob2.get_shape();
				ArrayList<Collision> col = s1.collision_detection(s2);
				//for collision i need: 
				/*
					whether there was a collision:
					a list of collisions:
						the point of collision
						the penetration or distance
						the normal of the surface involved
				*/
				if(col.size() > 0)
				{
					double n = (double)col.size();
					for(int k = 0; k < n; k++)
					{
						LObj.ll_pos_correct(ob1, ob2, col.get(k).dist, col.get(k).norm.cpy().inv());
						LObj.ll_resolve_collision(ob1, ob2, col.get(k).norm.cpy().inv(), col.get(k).point.cpy(), n);
					}
					
				}
			 }
		 }
	}

	
}
